dd-sdk-android/src/main/kotlin/com/datadog/android/core/internal/data/upload/DataFlusher.kt

```diff
@@ -0,0 +1,25 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.core.internal.data.upload
+
+import com.datadog.android.core.internal.net.DataUploader
+import com.datadog.android.core.internal.persistence.DataReader
+import com.datadog.android.core.internal.utils.sdkLogger
+
+internal class DataFlusher(
+    private val reader: DataReader,
+    private val dataUploader: DataUploader
+) {
+
+    fun flush() {
+        reader.readAll().forEach {
```

Comment:
you don't want to read all batch at once in memory. Depending on the batches size, this could lead to an OutOfMemory error. You can take a look at the `UploadWorker` class to see how we read them sequentially.